core/src/main/java/org/jruby/truffle/nodes/literal/ArrayLiteralNode.java

```diff
@@ -108,19 +108,24 @@ public RubyArray executeArray(VirtualFrame frame) {
                 try {
                     executedValues[n] = values[n].executeFloat(frame);
                 } catch (UnexpectedResultException e) {
-                    final Object[] executedObjects = new Object[n];
-
-                    for (int i = 0; i < n; i++) {
-                        executedObjects[i] = executedValues[i];
-                    }
-
-                    return makeGeneric(frame, executedObjects);
+                    return makeGeneric(frame, executedValues, n);
```

Comment:
As a general style guide: We should try to move exceptional parts of methods in extra helpers. Makes them easier to read and improves startup/warmup time (HotSpot compilers can avoid inlining a method, but the base line compiler would never throw away code and the optimizing ones only when it is *never* executed).